int32 num_IDs
int32 ID
int32[] path
